### PR TITLE
Add snapshot docs, fix formatting errors in a README and add missing file

### PR DESCRIPTION
**Pull request by sdodsley**
Add page for snapshot functionality usage.
Fix some formatting errors from last PR that were missed
Add the default `values.yaml` to the operator-k8s-plugin directory for consistency.

### CODE REVIEW

File: docs/flex-snapshot-for-flasharray.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+&emsp;_Output :_
+&emsp;None.
+&emsp;Exit code 0 means success, otherwise you will see an error message.
+
+### Running the snapshot CLI
+
+The snapshot CLI is deployed as a binary in Pure's dynamic provisioner pod. You do not need to download this binary to your computer. Instead you will use `kubectl exec` to run this binary.
+To create a snapshot you must get the name of the pure provisioner pod in the Kubernetes cluster. This name is randomly generated by Kubernetes and can be retrieved by running:
+```
+kubectl get -o name -l app=pure-provisioner pod | cut -d*/* -f2
```
Comment (trevormeiss) [inline docs/flex-snapshot-for-flasharray.md]: `cut -d*/*` returns `cut: bad delimiter` for me, but `cut -d/` and `cut -d '/'` both work

Comment (sdodsley) [reply docs/flex-snapshot-for-flasharray.md]: Fixed

File: docs/flex-snapshot-for-flasharray.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+### Running the snapshot CLI
+
+The snapshot CLI is deployed as a binary in Pure's dynamic provisioner pod. You do not need to download this binary to your computer. Instead you will use `kubectl exec` to run this binary.
+To create a snapshot you must get the name of the pure provisioner pod in the Kubernetes cluster. This name is randomly generated by Kubernetes and can be retrieved by running:
+```
+kubectl get -o name -l app=pure-provisioner pod | cut -d/ -f2
+```
+Now you can execute the snapshot command using the `<pure-podname>` just discovered:
+```
+kubectl exec <pure-podname> -- snapshot create -n <namespace> <pvc-name>
```
Comment (dsupure) [inline docs/flex-snapshot-for-flasharray.md]: snapshot is in pure-provisioner's bin folder, will it find the path directly?

File: docs/flex-snapshot-for-flasharray.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+&emsp;Example of a yaml file (e.g. pure-fa-snapshot-pvc.yaml)
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+ # Referenced in nginx-snapshot.yaml for the volume spec
+ name: pure-fa-snapshot-claim
+ annotations:
+**   snapshot.beta.purestorage.com/name: k8s-pvc-b9dd0972-c8b3-11e7-9ee8-fa163eb1e272.883661**
```
Comment (dsupure) [inline docs/flex-snapshot-for-flasharray.md]: Do we need to put two `**` here?

File: docs/flex-snapshot-for-flasharray.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+To delete a snapshot use the following command:
+```
+kubectl exec <pure-podname> -- snapshot delete <snapshotname>
+```
+
+### Examples
+
+#### Creating snapshots
+```
+# kubectl exec pure-provisioner-6d9878fd47-wp41b -- snapshot create -n k8s_ns1 pvc1
```
Comment (dsupure) [inline docs/flex-snapshot-for-flasharray.md]: Do you intend to put `#` in front of commands? I see all the code lines before this one don't have `#`.

File: docs/flex-snapshot-for-flasharray.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+kind: Pod
+metadata:
+ name: nginx-snapshot
+ namespace: default
+spec:
+ # Specify a volume that uses the claim defined in pure-fa-snapshot-pvc.yaml
+ volumes:
+ - name: pure-vol-snapshot
+   persistentVolumeClaim:
+**       claimName: pure-fa-snapshot-claim**
```
Comment (dsupure) [inline docs/flex-snapshot-for-flasharray.md]: same question here :).

File: docs/flex-snapshot-for-flasharray.md (hunk 2 above)
Comment (sdodsley) [reply docs/flex-snapshot-for-flasharray.md]: Yes

File: docs/flex-snapshot-for-flasharray.md (hunk 4 above)
Comment (sdodsley) [reply docs/flex-snapshot-for-flasharray.md]: The ones before are wrong :) Will fix

File: docs/flex-snapshot-for-flasharray.md (hunk 3 above)
Comment (sdodsley) [reply docs/flex-snapshot-for-flasharray.md]: Well caught - that's what you get for converting to markdown from a previous format - fixed

File: docs/flex-snapshot-for-flasharray.md (hunk 5 above)
Comment (pure-yesmat) [reply docs/flex-snapshot-for-flasharray.md]: This is still the same, was that on purpose?

Comment (sdodsley) [reply docs/flex-snapshot-for-flasharray.md]: Damn :)

**Pull request merged**